app/src/main/java/com/smashingboxes/epa_prototype_android/fitbit/FitbitOAuth2.java

```diff
@@ -0,0 +1,242 @@
+ */
+public class FitbitOAuth2 {
+
+    private static final String FITBIT_AUTH_URL = "https://www.fitbit.com/oauth2/authorize?";
+
+    private static final String FITBIT_REFRESH_AUTH_URL = "https://api.fitbit.com/oauth2/token";
+
+    private static final String CLIENT_ID = "229XJ8";
+
+    public enum ResponseType {
+        CODE, TOKEN
+    }
+
+    public enum Scope {
+        ACTIVITY, HEARTRATE, LOCATION, NUTRITION, PROFILE, SETTINGS, SLEEP, SOCIAL, WEIGHT
+    }
+
+    public enum Prompt {
+        NONE, LOGIN, CONSENT
+    }
+
+    public class FitbitAuthUriBuilder implements UriBuilder {
+
+        private ParamBuilder mParamBuilder;
+
+        @Override
+        public Uri getBaseRequestUri() {
+            return Uri.parse(FITBIT_AUTH_URL);
+        }
+
+        @Override
+        public Uri buildRequestUri() {
+            return Uri.parse(mParamBuilder.buildEncodedUri(getBaseRequestUri()));
+        }
+
+        /**
+         * Uses Implicit code since we shouldn't store o
+         * @param activity
+         * @return
+         */
+        @Override
+        public Uri buildDefault(Activity activity) {
+            FitbitAuthUriBuilder builder = new FitbitAuthUriBuilder();
+            builder.setClientId(CLIENT_ID).setResponseType(ResponseType.TOKEN)
+                    .setScopes(Arrays.asList(Scope.ACTIVITY, Scope.PROFILE, Scope.NUTRITION, Scope.PROFILE,
+                            Scope.SETTINGS, Scope.SLEEP, Scope.SOCIAL, Scope.WEIGHT))
+                    .setRedirectUri(FitbitOAuth2.this.getRedirectUriScheme(activity.getString(R.string.fitbit_scheme),
+                            activity.getString(R.string.fitbit_host)))
+                    .setExpiresIn();
+            return builder.buildRequestUri();
+        }
+
+        public FitbitAuthUriBuilder() {
+            mParamBuilder = new ParamBuilder();
+        }
+
+        public FitbitAuthUriBuilder setClientId(String clientId) {
+            mParamBuilder.put(RequestKeys.CLIENT_ID.getParamValue(), clientId);
+            return this;
+        }
+
+        public FitbitAuthUriBuilder setResponseType(@NonNull ResponseType responseType) {
+            mParamBuilder.put(RequestKeys.RESPONSE_TYPE.getParamValue(), responseType.name().toLowerCase());
+            return this;
+        }
+
+        public FitbitAuthUriBuilder setRedirectUri(@NonNull String uri) {
+            mParamBuilder.put(RequestKeys.REDIRECT_URI.getParamValue(), uri);
+            return this;
+        }
+
+        /**
+         * The list of fitbit scopes our application is interested in
+         *
+         * @param scopes
+         * @return
+         */
+        public FitbitAuthUriBuilder setScopes(@NonNull List<Scope> scopes) {
+            mParamBuilder.put(RequestKeys.SCOPE.getParamValue(), formatScopeList(scopes));
+            return this;
+        }
+
+        public FitbitAuthUriBuilder setExpiresIn() {
+            mParamBuilder.put(RequestKeys.EXPIRES_IN.getParamValue(), "86400");
```

Comment:
Just wanted to remind you of the hard-coded value here, but you may already know it.